File: shieldCore/src/main/java/com/dianping/agentsdk/framework/CellManagerInterface.java
package com.dianping.agentsdk.framework;

import android.view.ViewGroup;

import java.util.ArrayList;


/**
 * Created by hezhi on 15/12/11.
 */
public interface CellManagerInterface<T extends ViewGroup> {

    void setAgentContainerView(T containerView);

    //手动刷新接口
    void notifyCellChanged();

    void updateAgentCell(AgentInterface agent);

    void updateCells(ArrayList<AgentInterface> addList, ArrayList<AgentInterface> updateList, ArrayList<AgentInterface> deleteList);

}

File: shieldCore/src/main/java/com/dianping/agentsdk/framework/CellManagerInterface.kt
package com.dianping.agentsdk.framework

import android.view.ViewGroup

/**
 * Create By zhi.he at 2018/4/28
 */
interface CellManagerInterface<out T : @JvmWildcard ViewGroup> {
    fun setAgentContainerView(containerView: @UnsafeVariance T)

    //手动刷新接口
    fun notifyCellChanged()

    fun updateAgentCell(agent: AgentInterface)

    fun updateCells(
            addList: ArrayList<AgentInterface>?,
            updateList: ArrayList<AgentInterface>?,
            deleteList: ArrayList<AgentInterface>?
    )
}